src/core/qgismobileapp.cpp

```diff
@@ -528,49 +528,284 @@ void QgisMobileapp::loadProjectFile( const QString &path, const QString &name )
   if ( !fi.exists() )
     QgsMessageLog::logMessage( tr( "Project file \"%1\" does not exist" ).arg( path ), QStringLiteral( "QField" ), Qgis::Warning );
 
-  mAuthRequestHandler->clearStoredRealms();
+  const QString suffix = fi.suffix().toLower();
+  if ( suffix == QLatin1String( "qgs") || suffix == QLatin1String( "qgz") ||
+       suffix == QLatin1String( "gpkg" ) || suffix == QLatin1String( "shp" ) || suffix == QLatin1String( "kml" ) || suffix == QLatin1String( "kmz" ) || suffix == QLatin1String( "gpx" ) ||
+       suffix == QLatin1String( "tif" ) || suffix == QLatin1String( "pdf" ) || suffix == QLatin1String( "jpg" ) || suffix == QLatin1String( "png" ) )
```

Comment:
```suggestion
  QStringList supportedFileExtensions( {"qgs", "qgz", "gpkg", "shp", "kml", "kmz", "gpx", "tif", "pdf", "jpg", "png"} );
  if ( supportedFileExtensions.contains( suffix ) )
```

Reply:
I think this "supportedFileExtensions" can be somewhere more global and used also in the first comment